msi/jenkins.wxs

```diff
@@ -43,28 +43,47 @@
     <Upgrade Id="${MSI_PRODUCTCODE}">
       <UpgradeVersion Minimum="0.0.0" Maximum="99.99.99" Property="PREVIOUSVERSIONINSTALLED" />
     </Upgrade>
+
+    <CustomAction Id="BackupJenkinsXmlFile" Script="vbscript">
+      <![CDATA[
+        jenkinsDirPath = session.Property("JENKINSDIR")
+        Set fsObj = CreateObject("Scripting.FileSystemObject")
+
+        If fsObj.FolderExists(jenkinsDirPath) Then
+            jenkinsXmlPath = jenkinsDirPath + "jenkins.xml"
+
+            If fsObj.FileExists(jenkinsXmlPath) Then
+                Set shellObj = CreateObject("WScript.Shell")
+                cmdToExecute = "cmd.exe /C copy """ + jenkinsXmlPath + """ """ + jenkinsXmlPath + ".backup"" /Y"
```

Comment:
why are you executing a command here - this is brittle and you potentially have escaping hell going on.  (you can change the path where Jenkins is installed at install time!)

[hint `FileSystemObject.Copy()`](https://msdn.microsoft.com/en-us/library/6973t06a(v=vs.84).aspx)